libs/blog/shared/ui-icon/src/lib/icon/icon.component.ts

```diff
@@ -1,27 +1,22 @@
 import {
   ChangeDetectionStrategy,
   Component,
-  computed,
-  inject,
+  HostBinding,
   input,
 } from '@angular/core';
-import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
-
-import { getIcon, IconType } from '../icons';
 
 @Component({
```

Comment:
As we stick to avoid @HostBinding attributes i would do it like this:

```
@Component({
  selector: 'al-icon',
  standalone: true,
  template: '',
  changeDetection: ChangeDetectionStrategy.OnPush,
  styleUrls: ['icon.component.scss'],
  host: { '[style.-webkit-mask-image]': 'path()' },
})
export class IconComponent {
  name = input.required<string>();

  readonly path = computed(() => `url("assets/icons/${this.name()}.svg")`);
}

```

Reply:
It wouldn't work

Reply:
hmm it worked for me locally

Reply:
Done